DEVELOPMENT_GUIDE.md

```diff
@@ -0,0 +1,248 @@
+# Development setup guide for MLflow Website contributions
+
+This guide covers the general setup requirements that are needed to contribute to the MLflow website.
+
+The core of the MLflow website is powered by [Docusaurus](https://docusaurus.io/), a REACT-based static website generator framework.
+
+## Fork this repo
+
+In order to file a PR to this repository, you must file your PR from your fork of this repo. PRs that are submitted directly from the main branch will be closed.
```

Comment:
Not sure if we need to mention this because the only way for an external contributor to file a PR is to file a PR from their fork.